pmd-core/src/main/java/net/sourceforge/pmd/lang/Parser.java

```diff
@@ -32,10 +32,6 @@
      */
     TokenManager getTokenManager(String fileName, Reader source);
 
-    /**
-     * Indicates if this parser can actual parse, or if it can only tokenize.
-     */
-    boolean canParse();
```

Comment:
we also need to deprecate `Parser::canParse()` on master

Reply:
Not sure we should do that. On master it's still required to check this method, in the general case.

Reply:
I only see a usage in the old Designer (and the new Designer). We don't use it, instead we check, whether the language has a rule chain visitor....
So, I think, it's safe to deprecate it. If course, the language modules will still need to implement it.

Reply:
Done in 63e2454ed2